app/models/batch.rb

```diff
@@ -0,0 +1,36 @@
+class Batch < ApplicationRecord
+  has_and_belongs_to_many :checks
+
+  def to_h
+    {
+      id: id,
+      status: status,
+      links: checks.map(&:to_h),
+      totals: {
+        links: checks.count,
+        ok: checks.each.count { |check| check.status == "ok" },
+        caution: checks.each.count { |check| check.status == "caution" },
+        broken: checks.each.count { |check| check.status == "broken" },
+        pending: checks.each.count { |check| check.status == "pending" },
+      },
+      completed_at: completed_at,
+    }.deep_symbolize_keys
+  end
+
+  def started_at
+    checks.minimum(:started_at)
+  end
+
+  def completed_at
+    checks.maximum(:completed_at) if completed?
+  end
+
+  def completed?
+    checks.where(completed_at: nil).empty?
+  end
```

Comment:
These methods might all be running queries each time they're used. Might want to iterate through checks to find the items which would save queries. I'm not sure on this though, would need some experimenting.